pages/index.tsx

```diff
@@ -1,9 +1,21 @@
-import React from 'react'
+import { useSession } from 'next-auth/react'
+import LoginButton from '@/components/LoginButton'
 
-const DIDAdminPanel = () => (
-  <div>
-    <h1>DID Admin Panel</h1>
-  </div>
-)
+const DIDAdminPanel = () => {
+  const { data: session, status } = useSession()
+
+  return (
+    <div>
+      <h1>DID Admin Panel</h1>
+      {status === 'loading' && <p>Loading...</p>}
+      {status === 'unauthenticated' && <LoginButton />}
+      {status === 'authenticated' && (
+        <>
+          <p>Signed in as {session.user?.email}</p>
+        </>
+      )}
```

Comment:
If here you're validating if the user is unauthenticated, why is needed this logic?

https://github.com/Azordev/did-admin-panel/blob/b0660e259cbc9c9b4afa8bc028255306de16619e/components/LoginButton.tsx#L5-L19

Or it's for another case of use?

Reply:
Ah, the' session' use case is quite different from `status`. Please notice `status==='loading'` is not playing well with `if(session)`.

So the idea is: if you need to know when auth is loading, use status, but if loading is not a concern use `session`.